src/main/java/org/wso2/carbon/mongodb/query/MongoPreparedStatementImpl.java

```diff
@@ -0,0 +1,721 @@
+                JSONObject setObject = object.getJSONObject(key);
+                setUpdateObject(setObject.getJSONObject(MONGO_SET_FIELD));
+            } else {
+                if (parameterValue.containsKey(key)) {
+                    val = parameterValue.get(key);
+                    mapQuery.put(key, val);
+                }
+            }
+        }
+        this.query = new BasicDBObject(mapQuery);
+    }
+
+    /**
+     * Convert JSON Query to Aggregation Pipeline Object
+     *
+     * @param stmt JSONObject
+     */
+    private void getAggregationObjects(JSONObject stmt) {
+
+        Iterator<String> keys = stmt.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            if (key.equals(MONGO_COLLECTION_FIELD)) {
+                this.collection = db.getCollection(stmt.get(key).toString());
+            } else {
+                JSONObject value = stmt.getJSONObject(key);
+                if (key.equals(MONGO_LOOKUP_FIELD) || key.contains(MONGO_LOOKUP_SUB)) {
+                    if (isMultipleLookUp()) {
+                        mapLookUp = toMap(value);
+                    } else {
+                        mapLookUp = toMap(value);
+                        multiMapLookup.add(mapLookUp);
+                    }
+                } else if (key.equals(MONGO_PROJECT_FIELD)) {
+                    mapProject = toMap(value);
+                } else if (key.equals(MONGO_SORT_FIELD)) {
+                    mapSort = toMap(value);
+                } else if (key.equals(MONGO_GROUP_FIELD)) {
+                    mapGroup = toMap(value);
+                } else if (key.equals(MONGO_UNWIND_FIELD) || key.equals(MONGO_UNWIND_SUB)) {
+                    if (isMultipleLookUp()) {
+                        mapUnwind = toMap(value);
+                    } else {
+                        mapUnwind = toMap(value);
+                        multiMapUnwind.add(mapUnwind);
+                    }
+                } else {
+                    setMatchObject(value);
+                }
+            }
+        }
+    }
+
+    /**
+     * set object parameter with passed values
+     *
+     * @param stmt to set values
+     */
+    private void setMatchObject(JSONObject stmt) {
+
+        String[] elementNames = JSONObject.getNames(stmt);
+        for (String elementName : elementNames) {
+            Object value = stmt.get(elementName);
+            if (parameterValue.containsKey(elementName)) {
+                Object val = parameterValue.get(elementName);
+                if (!(value instanceof JSONObject) && !val.equals("%")) {
+                    mapMatch.put(elementName, val);
+                } else {
+                    if (value instanceof JSONObject) {
+                        JSONObject match = (JSONObject) value;
+                        String[] elements = JSONObject.getNames(match);
+                        for (String element : elements) {
+                            if (!val.equals("%")) {
```

Comment:
can we use a constant for "%"? avoid null pointer by changing the equals check as "%".equals(val)

Reply:
Resolved.